### PR TITLE
planner: tiny refactor around check index

### PR DESCRIPTION
**Pull request by winoros**
### What problem does this PR solve? 

Create `expression.Column` once. Then we don't need find column each time. 

### What is changed and how it works?

pass schema to `getGenExprs`

### Check List 

Tests 

 - Exsiting Unit test

### CODE REVIEW

File: planner/core/planbuilder.go [hunk 1]
```diff
@@ -823,21 +829,23 @@ func (b *PlanBuilder) buildPhysicalIndexLookUpReader(ctx context.Context, dbName
...
-			if c != nil {
-				col.Index = len(tblReaderCols)
-				tblReaderCols = append(tblReaderCols, c.ColumnInfo)
-				tblSchema.Append(&expression.Column{
-					ColName:  c.Name,
-					UniqueID: b.ctx.GetSessionVars().AllocPlanColumnID(),
-					RetType:  &c.FieldType})
-				colsMap[c.ID] = struct{}{}
-				if mysql.HasPriKeyFlag(c.Flag) {
+			info := findColumnInfoByID(tblInfo.Columns, col.ID)
```
Comment (lamxTyler) [inline planner/core/planbuilder.go]: Can we use `model.FindColumnInfo`?

File: planner/core/planbuilder.go [hunk 2]
```diff
@@ -823,21 +829,23 @@ func (b *PlanBuilder) buildPhysicalIndexLookUpReader(ctx context.Context, dbName
...
+			if info != nil {
+				tblReaderCols = append(tblReaderCols, info)
+				tblSchema.Append(col)
+				colsMap[col.ID] = struct{}{}
```
Comment (lzmhhh123) [inline planner/core/planbuilder.go]: Can we use `util.set.Int64Set`?

File: planner/core/planbuilder.go [hunk 3]
```diff
@@ -798,15 +802,17 @@ func (b *PlanBuilder) buildPhysicalIndexLookUpReader(ctx context.Context, dbName
 	schema := expression.NewSchema(make([]*expression.Column, 0, len(idx.Columns))...)
 	idxReaderCols := make([]*model.ColumnInfo, 0, len(idx.Columns))
 	tblReaderCols := make([]*model.ColumnInfo, 0, len(tbl.Cols()))
+	fullExprCols := expression.TableInfo2SchemaWithDBName(b.ctx, dbName, tblInfo)
+	genExprsMap, err := b.getGenExprs(ctx, dbName, tbl, idx, fullExprCols)
+	if err != nil {
+		return nil, errors.Trace(err)
```
Comment (lzmhhh123) [inline planner/core/planbuilder.go]: No need to trace?

File: planner/core/planbuilder.go (hunk 2 above)
Comment (winoros) [reply planner/core/planbuilder.go]: Sure, i forget that one.

File: planner/core/planbuilder.go (hunk 1 above)
Comment (winoros) [reply planner/core/planbuilder.go]: That one finds column by `name` not by ID

File: planner/core/planbuilder.go (hunk 3 above)
Comment (lonng) [reply planner/core/planbuilder.go]: Seems the backtrace has been added.

**Pull request merged**